src/utils/is-nil.ts

```diff
@@ -1,3 +1,35 @@
-export function isNil() {
+/**
+ * Checks if object is undefined or null. If memberAccessChain is provided it will traverse the object and check
+ * if the value associated to the memberAccessChain is undefined or null.
+ * @param object The object to check
+ * @param memberAccessChain The chain of access members of the object separated by the member access operator "."
+ * that will be traversed and checked if it's undefined or null.
+ * e.g. memberAccessChain = "a.b" will test whether object[a][b] is null or undefined.
+ */
+export function isNil(object: any, memberAccessChain?: string) {
+  if (_isNil(object)) {
+    return true;
+  }
 
+  if (_isNil(memberAccessChain) || memberAccessChain === '') {
+    return false;
+  }
+
+  const memberAccessOperator = '.';
+
+  let currentObject = object;
+  const keyChain = memberAccessChain.split(memberAccessOperator);
+
+  for (const key of keyChain) {
+    currentObject = currentObject[key];
+    if (_isNil(currentObject)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function _isNil(object) {
```

Comment:
1. "object" isn't a reserved word, but I feel like we should avoid it. Maybe try `value`?

2. Please add types:
    ```ts
    function _isNil(value: any): value is null | undefined { /* ... */ }
    ```